tests/tui/utils/compare_jsons.py

```diff
@@ -0,0 +1,67 @@
+import json
+import re
+
+
+def normalize_placeholders(path):
+    """
+    Replace equivalent placeholders in the given path.
+    """
+    placeholder_map = {"subject": "sub", "atlas": "desc", "map": "desc", "seed": "desc"}
+
+    for full, short in placeholder_map.items():
+        path = re.sub(rf"{{({full}|{short})}}", f"{{{short}}}", path)  # Normalize to correct equivalent
+
+    return path
+
+
+def load_and_normalize_json(file_path):
+    """
+    Load JSON from a file and normalize paths within it, while ignoring specified keys.
+    """
+    ignore_keys = {"halfpipe_version", "schema_version", "timestamp"}
+
+    with open(file_path, "r", encoding="utf-8") as f:
+        data = json.load(f)
+
+    if isinstance(data, list):
+        normalized_data = [normalize_entry(entry, ignore_keys) for entry in data]
+        return sorted(normalized_data, key=json.dumps)  # Sort for order independence
+
+    return normalize_entry(data, ignore_keys)
+
+
+def normalize_entry(entry, ignore_keys):
+    """
+    Normalize a single JSON entry, replacing placeholders in paths and removing ignored keys.
+    """
+    if isinstance(entry, dict):
+        return {
+            k: normalize_placeholders(v) if k == "path" and isinstance(v, str) else normalize_entry(v, ignore_keys)
```

Comment:
Why do the path patterns need to be modified for the comparison?

Reply:
because sometimes there can be different order of entries but that does not matter, so i reorder them and them compare. The different order is because old UI outputs something in different order that the new, that is why I put it here,
I think...

Reply:
That makes sense! Why normalize the path patterns though?

Reply:
because there are two interchangeable placeholders, for for examples, atlas you can have {map} but als {desc}, in the old ui when you set {map} the saved json had {desc}

Reply:
Yeah, I went with `desc` because that's what they call in in the BIDS standard. I think we should keep them as is. After all, the new UI is supposed to create a fully compatible `spec.json`, right?

Reply:
yes, that is why in the test i have this path thing, so that when one file uses that and the other one that they will be evaluated as the same

Reply:
Why don't they both use the same one?

Reply:
Does that work when running the workflows?

Reply:
not tested, but it should, aren't these after all interchangable placeholders?